### ISSUE TITLE
Flake in test_create_gate_set_validator

### ISSUE THREAD

**Issue by 95-martin-orion** [quantumlib/Cirq#4796]
**Description of the issue**
`cirq-google/cirq_google/engine/engine_validator_test.py::test_create_gate_set_validator` experiences a flake in ~0.4% of tests where the generated circuit is too long. [Example failure](https://github.com/quantumlib/Cirq/runs/4705935573?check_suite_focus=true)

**How to reproduce the issue**

```
pip install pytest-repeat
pytest cirq-google/cirq_google/engine/engine_validator_test.py -k create_gate_set -p no:warnings --count=1000
# ~0.4% flake with "RuntimeError: INVALID_PROGRAM: Program too long."
```

**Cirq version**

0.14.0.dev

**Comment by tanujkhattar** [comment]
From cirq sync: 

This is testing that really large circuits are invalidated. We should make the test deterministic to fix this. The test needs to be there.

**Comment by dstrain115** [comment]
This was fixed in #4908.
